### ISSUE TITLE
Error when navigating with Vue Router

### ISSUE THREAD

**Issue by ChadTaljaardt** [vuejs/vue-router#1728]
### Version
2.7.0

### Reproduction link
[https://jsfiddle.net/chrisvfritz/50wL7mdz/](https://jsfiddle.net/chrisvfritz/50wL7mdz/)

### Steps to reproduce
To be honest i don't actually know how to reproduce this due to it being intermittent. For this reason i cannot provide a reproduction of the error. 

### What is expected?
When navigating between panels, the system should navigate smoothly. 

### What is actually happening?
A error gets thrown in the console and vue wont navigate until i refresh the browser.

---
```
TypeError: Cannot read property '__ob__' of undefined
    at a.t.$destroy (vue.esm.js:2471)
    at destroy (vue.esm.js:3546)
    at y (vue.esm.js:5274)
    at y (vue.esm.js:5279)
    at a.t.nodeOps [as __patch__] (vue.esm.js:5560)
    at a.t.$destroy (vue.esm.js:2477)
    at destroy (vue.esm.js:3546)
    at y (vue.esm.js:5274)
    at y (vue.esm.js:5279)
    at y (vue.esm.js:5279)
S @ vue.esm.js:566
(anonymous) @ vue.esm.js:703
t @ vue.esm.js:648
Promise resolved (async)
e @ vue.esm.js:663
(anonymous) @ vue.esm.js:711
Ot @ vue.esm.js:2811
Bo.update @ vue.esm.js:2948
Mo.notify @ vue.esm.js:778
set @ vue.esm.js:1000
(anonymous) @ vue-router.esm.js:2390
(anonymous) @ vue-router.esm.js:2389
te.updateRoute @ vue-router.esm.js:1924
(anonymous) @ vue-router.esm.js:1802
(anonymous) @ vue-router.esm.js:1911
r @ vue-router.esm.js:1649
r @ vue-router.esm.js:1656
r @ vue-router.esm.js:1656
at @ vue-router.esm.js:1660
(anonymous) @ vue-router.esm.js:1906
r @ vue-router.esm.js:1649
(anonymous) @ vue-router.esm.js:1653
(anonymous) @ vue-router.esm.js:1891
(anonymous) @ vue-router.esm.js:1727
p @ vue-router.esm.js:1870
r @ vue-router.esm.js:1652
r @ vue-router.esm.js:1656
r @ vue-router.esm.js:1656
r @ vue-router.esm.js:1656
r @ vue-router.esm.js:1656
r @ vue-router.esm.js:1656
at @ vue-router.esm.js:1660
te.confirmTransition @ vue-router.esm.js:1899
te.transitionTo @ vue-router.esm.js:1801
e.push @ vue-router.esm.js:2173
ie.push @ vue-router.esm.js:2416
M @ vue-router.esm.js:404
e @ vue.esm.js:1814
```

**Comment by posva** [comment]
Sorry, but we need a reproduction to take a look. If you ever manage to reproduce it, make sure to open a new issue. For the moment I recommend you to look for help on the [forum](http://forum.vuejs.org/) or the [Discord server](https://vue-land.js.org/)

**Comment by PeterMaquiran** [comment]
did you guys found any solution

**Comment by carlosIntrigue** [comment]
I'm having the same issue right now

**Comment by carlosIntrigue** [comment]
I solved it after looking at #1785
One of my components data function didn't return an object
